src/level/robotmain.cpp

```diff
@@ -5866,3 +5890,28 @@ bool CRobotMain::GetDebugCrashSpheres()
 {
     return m_debugCrashSpheres;
 }
+
+void CRobotMain::PushToCommandHistory(std::string str)
+{
+    if (!m_commandHistory.empty() && m_commandHistory.front() == str) // already in history
+        return;
+
+    m_commandHistory.push_front(str);
+
+    if (m_commandHistory.size() > 50) // to avoid infinite growth
+        m_commandHistory.pop_back();
+}
+
+std::string CRobotMain::GetNextFromCommandHistory()
+{
+    if (m_commandHistory.empty() || (int)m_commandHistory.size() <= m_commandHistoryIndex + 1) // no next element
```

Comment:
And if you look in the Jenkins logs, you'll see this:
```
/var/lib/jenkins/workspace/colobot/colobot/PR-869/src/level/robotmain.cpp: In member function 'std::string CRobotMain::GetNextFromCommandHistory()':
/var/lib/jenkins/workspace/colobot/colobot/PR-869/src/level/robotmain.cpp:5906:64: warning: use of old-style cast [-Wold-style-cast]
     if (m_commandHistory.empty() || (int)m_commandHistory.size() <= m_commandHistoryIndex + 1) // no next element
                                                                ^
```
So you just introduced a warning while fixing another warning ;)